engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -217,13 +296,32 @@ public function get($guid, $type = '') {
 			return false;
 		}
 
-		// Check local cache first
-		$new_entity = _elgg_services()->entityCache->get($guid);
-		if ($new_entity) {
+		$guid = (int) $guid;
+
+		$memcache = _elgg_get_memcache('new_entity_cache');
+		
+		$entity = $this->entity_cache->get($guid);
+		if (!$entity) {
+			$entity = $memcache->load($guid);
+		}
+
+		if (!$entity instanceof \ElggEntity) {
+			$entity = false;
+		}
+
+		// Validate accessibility
+		if ($entity && !elgg_get_ignore_access() && !has_access_to_entity($entity)) {
```

Comment:
Users would be good to have in there, since every logged-in boot has to fetch one and there's lots of user prefetching. The user prefetcher could try memcache first...